src/char/int_storage.c

```diff
@@ -240,13 +392,126 @@ int mapif_save_guild_storage_ack(int fd, int account_id, int guild_id, int fail)
 	return 0;
 }
 
+//=========================================================
+// Account Storage
 //---------------------------------------------------------
-// packet from map server
+/**
+ * Parses account storage load request from map server.
+ * @packet 0x3010 [in] <account_id>.L
+ * @param  fd     [in] file/socket descriptor
+ * @return 1 on success, 0 on failure.
+ */
+int mapif_parse_AccountStorageLoad(int fd)
+{
+	int account_id = RFIFOL(fd, 2);
+
+	Assert_ret(fd > 0);
+	Assert_ret(account_id > 0);
+
+	mapif->account_storage_load(fd, account_id);
+
+	return 1;
+}
+
+/**
+ * Loads the account storage and send to the map server.
+ * @packet 0x3805     [out] <account_id>.L <struct item[]>.P
+ * @param  fd         [in]  file/socket descriptor.
+ * @param  account_id [in]  account id of the session.
+ * @return 1 on success, 0 on failure.
+ */
+int mapif_account_storage_load(int fd, int account_id)
+{
+	struct storage_data *stor = NULL;
+	int count = 0, i = 0, len = 0;
+
+	Assert_ret(account_id > 0);
+
+	stor = (struct storage_data *) idb_ensure(inter_storage->account_storage, account_id, inter_storage->ensure_account_storage);
+
+	count = inter_storage->fromsql(account_id, stor);
+
+	len = 8 + count * sizeof(struct item);
+
+	WFIFOHEAD(fd, len);
+	WFIFOW(fd, 0) = 0x3805;
+	WFIFOW(fd, 2) = (uint16) len;
+	WFIFOL(fd, 4) = account_id;
+	for (i = 0; i < count; i++)
+		memcpy(WFIFOP(fd, 8 + i * sizeof(struct item)), &VECTOR_INDEX(stor->item, i), sizeof(struct item));
+	WFIFOSET(fd, len);
+
+	return 1;
+}
 
+/**
+ * Parses an account storage save request from the map server.
+ * @packet 0x3011 [in] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @param  fd     [in] file/socket descriptor.
+ * @return 1 on success, 0 on failure.
+ */
+int mapif_parse_AccountStorageSave(int fd)
+{
+	int payload_size = RFIFOW(fd, 2) - 8, account_id = RFIFOL(fd, 4);
+	int i = 0, count = 0;
+	struct storage_data *cp_stor = NULL, p_stor = { 0 };
+
+	Assert_ret(fd > 0);
+	Assert_ret(account_id > 0);
+
+	cp_stor = (struct storage_data *) idb_ensure(inter_storage->account_storage, account_id, inter_storage->ensure_account_storage);
+
+	count = payload_size/sizeof(struct item);
+
+	VECTOR_INIT(p_stor.item);
+
+	if (count > 0) {
```

Comment:
I went ahead and added more than the required number of lines into the clause. Fixed and added.